src/wymeditor/core.js

```diff
@@ -944,6 +1022,44 @@ WYMeditor._computeBasePath = function (wymPath) {
     return wymPath.substr(0, lastSlashIndex + 1);
 };
 
+WYMeditor._getSkinUIOverrides = function (providedOptions, skin) {
```

Comment:
> this static method may be easier to understand if it also did the merging and not only provide the object for the merge.

I think limiting it to just providing the objects (doing exactly one thing) makes it easier to test and understand. It all supports the composability desired to perform the multi-step merge. Why have a function that does two things when it's useful doing one thing?